ister_gui.py

```diff
@@ -641,24 +643,215 @@ def validate_condition(self):
+        except Exception:
+            return False
+        # pylint: enable=E1135
+
+    def _static_configuration(self, _):
+        """
+        Writes the configuration on /etc/systemd/network/10-en-static.network
+        """
+        try:
+            ipaddress.ip_address(self.static_ip.get_edit_text())
+        except:
+            # the main loop is waiting on this method to exit so it can report
+            # the error
+            raise urwid.ExitMainLoop()
+
+        if self.interface.get_edit_text() not in self.ifaceaddrs.keys():
+            # the main loop is waiting on this method to exit so it can report
+            # the error
+            raise urwid.ExitMainLoop()
+
+        path = '/etc/systemd/network/'
+        if not os.path.exists(path):
+            os.makedirs(path)
+
+        with open(path + "10-en-static.network", "w") as file:
+            file.write("[Match]\n")
+            file.write("Name={}\n\n".format(self.interface.get_edit_text()))
+            file.write("[Network]\n")
+            file.write("Address={0}\n".format(self.static_ip.get_edit_text()))
+            file.write("Gateway={0}\n".format(self.gateway.get_edit_text()))
+
+        try:
+            subprocess.call(['systemctl', 'restart',
+                             'systemd-networkd', 'systemd-resolved'])
+        except Exception as err:
+            Alert('Error!', err).do_alert()
+
+        raise urwid.ExitMainLoop()
+
+    def _reset_network(self, _):
+        """Reset the network to original configuration by removing static
+        network file and restarting network services"""
+        try:
+            os.remove('/etc/systemd/network/10-en-static.network')
+        except FileNotFoundError:
+            pass
+
+        try:
+            subprocess.call(['systemctl', 'restart', 'systemd-networkd'])
+        except:
+            raise urwid.ExitMainLoop()
+
+        # give the network a chance to start up again
+        time.sleep(.1)
+        raise urwid.ExitMainLoop()
+
+    def _set_proxy(self, _):
+        """Set the user defined proxy for the installer in the template"""
+        if self.installer_proxy.get_edit_text():
+            self.config['Proxy'] = self.installer_proxy.get_edit_text()
+
+        raise urwid.ExitMainLoop()
+
+    def _find_interface_ip(self):
+        """Find active interface and ip address"""
+        # pylint: disable=E1103
+        addrs = {}
+        af_inet = netifaces.AF_INET
+        for if_name in netifaces.interfaces():
+            ifaddrs = netifaces.ifaddresses(if_name)
+            if af_inet in ifaddrs:
+                ip_addrs = ifaddrs[af_inet][0]
+                if ip_addrs['addr'] and '127.0.' not in ip_addrs['addr']:
+                    addrs[if_name] = ip_addrs['addr']
+
+        # pylint: enable=E1103
+        self.ifaceaddrs = addrs
+        for interface in addrs:
+            if interface.startswith('e'):
+                return (interface, addrs[interface])
```

Comment:
So the above adds a lot of non gui functionality to ister_gui.py and I'm worried we won't have unit test coverage for it just the expect scripts. I'm starting to think an ister python library may be the best place to put some of these features unless you want to extend ister_test.py to import the gui for testing. Up to you on what you think would be best.